### DISCUSSION TITLE
How to invert boolean values? (not operator)

### DISCUSSION THREAD

**Discussion by KSXGitHub** [nushell/nushell#7275] [Q&A] [answered]
I have this data:

| name  | is_male |
|-------|---------|
| Bob   | true    |
| Alice | false   |

I want to make this table:

| name  | is_male | is_female |
|-------|---------|-----------|
| Bob   | true    | false     |
| Alice | false   | true      |

This code doesn't work because the `not` operator does not exist:

```nu
open data.json | insert is_female {|x| not $x.is_male}
```

**Comment by fdncred** [comment, marked as answer]
It seems to work for me. What version of nushell are you on?
![image](https://user-images.githubusercontent.com/343840/204551280-0b30dcdd-e48f-4085-859f-e2d77dedf30f.png)

This is the version I'm on.
|key|value|
|-|-|
|version|0.72.0|
|branch|main|
|commit_hash|9c800bcb2cfac71931d8485d146087c943d108d7|
|build_os|windows-x86_64|
|build_target|x86_64-pc-windows-msvc|
|rust_version|rustc 1.65.0 (897e37553 2022-11-02)|
|rust_channel|1.65.0-x86_64-pc-windows-msvc|
|cargo_version|cargo 1.65.0 (4bc8f24d3 2022-10-20)|
|pkg_version|0.72.0|
|build_time|2022-11-29 07:12:42 -06:00|
|build_rust_channel|release|
|features|database, dataframe, default, trash, which, zip|
|installed_plugins|custom-value generate, custom-value generate2, custom-value update, from parquet, gstat, inc, nu-example-1, nu-example-2, nu-example-3, query, query json, query web, query xml|

**Reply by KSXGitHub** [reply]
It was a mistake on my part. `not` does not seems to appear on [the "Command Reference" of the book](https://www.nushell.sh/book/command_reference.html).

**Reply by fdncred** [reply]
It's not a command and technically not an operator. It's an expression I guess.

**Reply by KSXGitHub** [reply]
I even edit the link to point to `not` explicitly, but it's 404: https://www.nushell.sh/book/commands/not.html

**Reply by KSXGitHub** [reply]
> It's not a command and technically not an operator. It's an expression I guess.

But its form looks not different than a command!

**Reply by fdncred** [reply]
This is new for the release that's going out today. Hopefully it'll help.
![image](https://user-images.githubusercontent.com/343840/204552056-90f66b01-8421-4f9a-a97a-5a5ac600971e.png)

**Reply by theAkito** [reply]
Wow, this operator exists? I thought it doesn't exist. It's really hard to find a common word like `not` in the online command documentation...

**Reply by theAkito** [reply]
> This is new for the release that's going out today. Hopefully it'll help. ![image](https://user-images.githubusercontent.com/343840/204552056-90f66b01-8421-4f9a-a97a-5a5ac600971e.png)

Looks great, thank you!
